<p>Tags : #Redaction In RIDGE, there is two set of summary statistics that are used : the basic set, which is used to describe a single locus and an extended set which is use to describe an ensemble of loci.  The basic set first :  - <strong>bialsite</strong> : number of bi allele sites in the locus<br />
- <strong>Fst</strong> : Fst estimation using the method of Hudson (1992) elaborated by Bhatia et al. (2013). - <strong>divAB</strong> : Estimate nucleotide divergence between two populations (dxy) within a given region, which is the average proportion of sites (including monomorphic sites not present in the data) that differ between randomly chosen pairs of chromosomes, one from each population. - <strong>netDivAB</strong> : nucleotide divergence without ancestral polymorphism (Da). The ancestral polymorphism is asume to mean the mean of population polymorphism. Da = Dxy - (A+B )/2 - <strong>piA , piB</strong> : Estimate nucleotide diversity  within a given region, which is the average proportion of sites (including monomorphic sites not present in the data) that differ between randomly chosen pairs of chromosomes. - <strong>thetaA , thetaB</strong> : Estimate nucleotide diversity with watterson theta - <strong>DtajA, DtajB</strong> : Tajima’s D - <strong>ss</strong> : proportion of polymorphism shared between population - <strong>sf</strong> : proportion of polymorphism with fixed difference between population - <strong>sxA, sxB</strong> : proportion of polymorphism specific to a population [[joint_sfs_explanation.jpg]] <strong>Fig 1</strong> :<em>Joint sfs spectrum representation with explanation on where to find ss(central orange zone), sf (blue zone), sxA(red zones) and sxB (green zones). To calculate them, it is the sum of the number of SNP inside the zone and divide by the total number of SNP.</em></p>
<p>To describe the ensemble of loci, the mean (avg) , the standard error (std)  and the median (median) is calculated for each previously described summary statistics. In addition new variable describe : - <strong>pearson_r_pi</strong> : pearson correlation between the pi of each population. - <strong>pearson_r_theta</strong> : pearson correlation between the Watterson’s theta of each population. - <strong>pearson_r_divAB_netdivAB</strong> : pearson correlation between the dxy and Da - <strong>pearson_r_divAB_FST</strong>: pearson correlation between dxy and FST - <strong>pearson_r_netdivAB_FST</strong>: pearson correlation between Da and FST - <strong>ss_sf</strong> : number of loci which contain shared polymorphism and fixed difference - <strong>ss_noSf</strong> : number of loci which contain shared polymorphism and ==no== fixed difference - <strong>noSs_sf</strong> : number of loci which contain ==no== shared polymorphism and fixed difference - <strong>noSs_noSf</strong> : number of loci which contain ==no== shared polymorphism and ==no== fixed difference</p>
<p>All the computation part is done by <a href="%5Bhttps://github.com/cggh/scikit-allel%5D(https://github.com/cggh/scikit-allel)"><strong>scikit-allel</strong></a> and <a href="https://github.com/numpy/numpy"><strong>numpy</strong></a> packages.</p>
